### ISSUE TITLE
Blacklist reactions from issue comments

### ISSUE THREAD

**Issue by tziporaziegler** [isaacs/github#967]
Since the introduction of [GitHub reactions][1], users can show their support or disagreement of posts by simply adding a reaction, and they no longer need to add a comment of their own. This improvement is great as it helps keep discussions focused, by decreasing the amount of frivolous comments.

Too often, however, comment threads are still flooded with comments that look like the following:
+1
-1
👍
👎

These comments are simply worthless and annoying:
- They disrupt the flow of the conversation.
- They cause *useful* comments to be overlooked, as you need to scroll through endless amounts of *useless* comments to pick out the few that actually add something of importance. 
- Each time a users posts such a comment, everyone who is subscribed to the thread is spammed with a worthless email.  I know I've personally started ignoring GitHub emails more, since more often than not, opening the email is just a waste of time. I've also encountered numerous users on different posts, who wanted to stay updated on an issue, yet unsuscribed, just to stop the annoying flood of emails.

This is especially an issue on popular issue requests that have hundreds of subscribers. Every time someone leaves a comment like this, instead of just using the reaction buttons, hundreds of users are bothered with an email that doesn't add anything to the conversation.

If you need an example, just look at the following issue discussion thread, where the number of useless comments amounts to several hundred: https://github.com/isaacs/github/issues/253
If you want to get a feel of what I am referring to, just subscribe to any of the popular threads in this repo, and watch the flood of useless "+1" emails creep into your inbox.

I am therefore adding a feature request to blacklist all comments that should really just be a reaction. If a user attempts to post a comment with just a reaction, their post should be blocked, and they should instead be prompted with a message like:
>+1, -1, 👍, 👎 are not allowed as comments. Use [reactions][1] instead!

Bonus points if the following comments were also blacklisted:
+2
+1000
+9999
+WhateverRandomNumberSomeoneUses

[1]: https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments

**Comment by QuantumInformation** [comment]
Great idea.

**Comment by cirosantilli** [comment]
Users will start doing: ":+1: some trash" :-)

**Comment by JayFoxRox** [comment]
@cirosantilli I don't think that's a valid concern. Those who actively want to disrupt conversations will *always* find a way to do so.

This is mostly about keeping idiots out. By rejecting their first attempt they will be made aware of their stupidity.
And I think what OP proposed is a good way of doing this.

That said, a possible strategy to implement this, is to filter if the 👍 or 👎 is the first or last symbol in a post, then warn the user before allowing their comment. Posts entirely consisting of emojis / reactions could be rejected entirely.

**Comment by tziporaziegler** [comment]
@cirosantilli I think a lot of users are just not aware that their "+1" comments are unwanted. I also think that some users don't realize they should be using reactions.

Like @JayFoxRox said, this won't stop users who purposely want to add useless comments. It should, however, limit the amount of useless comments by stopping people who want to innocently agree with another comment and think the way to do that is adding a "+1" comment, and by encouraging people to use reactions more.

I also think this is a better proposal than just automatically turning the comments into reactions, because that will just cause users to wonder where there comment disappeared to. I also think this method with give users a clearer message that only comments that add something of value is wanted.

**Comment by cirosantilli** [comment]
I'm joking.

**Comment by TPS** [comment]
Duplicates _(my 😉)_ #640, right?

**Comment by tziporaziegler** [comment]
@TPS It is definitely related since it is the same issue, but we are proposing two different solutions.  I don't think the comments should be automatically converted to reactions. I think that is confusing. Instead, I think users should be prompted with an error message when they try to post and their comment should be blocked.

I mentioned the difference in a comment for issue https://github.com/isaacs/github/issues/968, which is a duplicate of yours.

**Comment by TPS** [comment]
I agree, your solution is an elegant _addition_ to mine (though, I think, in implementation, there isn't a lot of difference between _conversion_ & _blocking_).

**Comment by JayFoxRox** [comment]
Only because issues touch a similar topic or area, does not mean that they are duplicates.
In this case, it's not even related. It's not an "*addition*" to your solution. It's an entirely different approach altogether.

This issue:

- Well defined scope of what should happen (yours leaves a ton of room for intepretation, see https://github.com/isaacs/github/issues/968#issuecomment-315405653)
- Does not change stuff *actively* (as in: changing what people say) but is a *passive* moderation aid (as in: preventing them from doing something bad)
- It does nothing "retroactively", which seems to be a key argument in your issue

I'd support this, but I dislike #968 and #640 .
Wether the other ones are *also* implemented (entirely independent), is a different matter.
However, for now, this issue has a far better chance in being implemented first: it's much more specific and there is little room left to discuss (which is good).
Turning this into an "addition" for any of the other issues would delay it with discussions. It would also possibly bend it until it's not even doing what it was intended to do anymore.

**Comment by jack-guy** [comment]
Just to add my two cents to this - I'm finding myself constantly forced to chose between staying in-the-know on issues I care about and being inudated with +1s and the like in emails and notifications. Increasingly I'm finding myself unsubscribing from these threads. GitHub is full of smart people but that doesn't give it a pass for not having UI that guides users towards understanding, especially for something as simple as reactions.

**Comment by gregkonushev-okta** [comment]
+1

**Comment by mspiegel** [comment]
I've implemented this feature as a GitHub App. Looking for beta-testers. Check out https://github.com/apps/always-be-closing if you're interested. Any feedback would be appreciated at https://github.com/thoughtdealership/abc-users.
